apps/example-app/src/app/examples/04-forms-with-material.spec.ts

```diff
@@ -37,12 +40,46 @@ test('is possible to fill in a form and verify error messages (with the help of
 
   expect(nameControl).toHaveValue('Tim');
   expect(scoreControl).toHaveValue(7);
+  expect(colorControl).toHaveTextContent('Green');
 
   const form = screen.getByRole('form');
   expect(form).toHaveFormValues({
     name: 'Tim',
     score: 7,
   });
-
   expect((fixture.componentInstance as MaterialFormsComponent).form?.get('color')?.value).toBe('G');
 });
+
+test('is should show pre-set form values', async () => {
+  const formBuilder = new FormBuilder();
+
+  const { fixture, detectChanges } = await render(MaterialFormsComponent, {
+    imports: [MaterialModule],
+    componentProperties: {
```

Comment:
Thanks for enhancing this example.
I would like to change a few things, the most important one to not overwrite the form.
Because this also overwrites the validators, which can be problematic if requirements change.

It's also sufficient to invoke a change detection cycle (we don't need to click on the trigger).

Thoughts?

```ts
  const { fixture, detectChanges } = await render(MaterialFormsComponent, {
    imports: [MaterialModule],
  });

  fixture.componentInstance.form.setValue({
    name: 'Max',
    score: 4,
    color: 'B'
  })
  detectChanges();
  
  const nameControl = screen.getByLabelText(/name/i);
  const scoreControl = screen.getByRole('spinbutton', { name: /score/i });
  const colorControl = screen.getByRole('combobox', { name: /color/i });

  expect(nameControl).toHaveValue('Max');
  expect(scoreControl).toHaveValue(4);
  expect(colorControl).toHaveTextContent('Blue');
```

Reply:
@timdeschryver yes, that is indeed a very good point. I adapted it.

Reply:
While this works, I'm thinking if we can do something about this to provide a better experience.
Not sure if this is better though 😅
If you have an idea, feel free to drop it here.

```ts
const { fixture } = render(Component);
// this would be new and invokes a CD cycle 
invoke(() => {
  fixture.componentInstance.form.setValue();
})
```

Reply:
@timdeschryver If we had to update the form values from within the test, this might be an option.

However, in a scenario where one uses custom written Angular components for different form fields, there is most likely a `formControl` being passed into it through its input properties. If we now follow the example from https://github.com/testing-library/angular-testing-library/blob/main/apps/example-app/src/app/examples/02-input-output.spec.ts#L10 how to pass input properties, one would not think of having to call `detectChanges()` to validate any pre-set value within the `formControl`.

Reply:
Those examples are based on "real-world" scenarios.
I didn't had the need (yet) to pass form controls to child components - and editing the code just for the tests doesn't feel right. The same problem also applies when validators are added to the control.

For now, I think this example shows a user how to do it the best way.
We can revisit and modify this later if we can think of a better solution.